api/app/Http/Controllers/UserController.php

```diff
@@ -18,17 +56,335 @@
+        
+        /** @var UserLaravel|null $user */
+        $user = UserLaravel::query()->find((int)$id);
+        if (is_null($user)) {
+            return response()->json(new ErrorResponse(Response::$statusTexts[404]), 404);
+        }
+        
+        $request->validate([
+            'phoneNumber' => 'string|nullable',
+            'socialNetworkUrl' => 'string|nullable',
+            'isBlocked' => 'boolean|nullable',
+            'apiKey' => 'string|nullable',
+            'notes' => 'string|nullable',
+            'name' => 'string|nullable',
+        ]);
+        
+        if ($currentUser->roleId !== $this->roles->Admin) {
+            $attributes = $request->only([
+                'phoneNumber',
+                'socialNetworkUrl',
+                'name'
+            ]);
+        } else {
+            $attributes = $request->only([
+                'phoneNumber',
+                'socialNetworkUrl',
+                'isBlocked',
+                'apiKey',
+                'notes',
+                'name'
+            ]);
+        }
+        
+        $user->update($attributes);
+        
+        if (
+            ($currentUser->roleId === $this->roles->Admin) &&
+            ((bool)$request->input('isBlocked') === true)
+        ) {
+            $user->tables->each(function (TableLaravel $table) {
+                $table->generators->each(function (GeneratorLaravel $generator) {
+                    $generator->lastGeneration = $this->xmlGenerationService
+                        ->getEmptyGeneratedXML($generator->targetPlatform);
+                    $generator->update();
+                });
+            });
+        }
+        
+        return response()->json(null, 200);
+    }
+    
+    /**
+     * Put /users/{$id}/token
+     *
+     * @param Request $request
+     * @param         $id
+     *
+     * @return JsonResponse
+     * @throws Exception
+     */
+    public function refreshToken(Request $request, $id): JsonResponse
+    {
+        /** @var UserLaravel $currentUser */
+        $currentUser = auth()->user();
+    
+        Log::channel('apilog')->info('PUT /users/'.$id.'/token - user '.$currentUser->id);
+        
+        if ($currentUser->roleId !== $this->roles->Admin) {
+            return response()->json(new ErrorResponse(Response::$statusTexts[403]), 403);
+        }
+        
+        /** @var UserLaravel|null $user */
+        $user = UserLaravel::query()->find((int)$id);
+        if (is_null($user)) {
+            return response()->json(new ErrorResponse(Response::$statusTexts[404]), 404);
+        }
+        
+        if ($user->roleId === $this->roles->Admin) {
+            return response()->json($user->apiKey, 200);
+        }
+        
+        $newApiKey = md5(Guid::uuid4()->toString());
+        $user->apiKey = $newApiKey;
+        $user->update();
+        
+        return response()->json($newApiKey, 200);
+    }
+    
+    /**
+     * Put /users/update
+     * Update user to Login Pass
```

Comment:
Why is it a separate method and why this route is so weird?

Also, PUT should normally be idempotent and as I can see, this endpoint can be called only once per user, so POST should be used instead.